toml.md

```diff
@@ -104,27 +104,33 @@ Keys
 
 A key may be either bare, quoted, or dotted.
 
-**Bare keys** may only contain ASCII letters, ASCII digits, underscores, and
-dashes (`A-Za-z0-9_-`). Note that bare keys are allowed to be composed of only
-ASCII digits, e.g. `1234`, but are always interpreted as strings.
+**Bare keys** may contain any letter-like Unicode character from any Unicode script,
+may contain ASCII digits, dashes and underscores. Punctuation, spaces, arrows, box drawing 
+and private use characters are not allowed.  Note that bare keys are allowed to be
```

Comment:
Could we explicitly list the allowed Unicode codepoint ranges similarly as we already do for comments and strings? I think that is valuable to implementers.

Reply:
Personally I'd rather see this defined in Unicode categories rather than codepoints (both in the ABNF and in here – you'll still have to copy the codepoints to the ABNF as I don't think you can list Unicode categories, but that seems like a better reference point than a series of characters).

Reply:
@hukkin, I wouldn't mind doing that, but it makes the paragraph a little wordier and arguably a bit harder to read. I strove to keep it simple ;).

Reply:
I understand, but my argument is that for an implementer it'd be a lot _easier_ to read. Compare that to Google searching the meaning the vague concepts like "letter-like Unicode character", "Unicode arrows" and so on. I think a technical spec shouldn't leave room for interpretation.

Reply:
I tend to agree with @hukkin here, though would also be satisfied with an explicit note directing implementers to the ABNF, if the full list is too verbose.

Reply:
@hukkin & @marzer, let me think of some useful wording and see how it pans out.

Reply:
@hukkin, @marzer, done. I chose to use wording like "all of this, except ...", as that tended to be more readable in my opinion. The ABNF doesn't have except-syntax (EBNF does, btw), so the ABNF looks a bit more convoluted than it needs to be, given that certain individual chars need to be excluded.

The prose is as follows in the `toml.md` now:

* From the ASCII characters, only A-Z, a-z, 0-9, _ and - are allowed
* From the rest of the first 256 characters, only 0080-00BF, "×" (00D7) and "÷" (00F7) are disallowed
* All of 0100-1FFF are allowed, except ";" (037E)
* Characters 200C, 200D, 203F, 2040, 2070-218F, 2C00-2FEF are allowed
* All characters from 3001, except surrogates (D800-D999), Private Use (E000-F8FF, F0000-100000) and process-internal use (FDD0-FDEF)

Reply:
Thanks! I'm happy with this, but I think I'd slightly prefer the type of language ChristianSi used in this comment https://github.com/toml-lang/toml/pull/891#pullrequestreview-911367703

- It is more in line with the language used defining allowed comment and string characters
- You already specify the allowed characters in "human language" at the start of the paragraph. This listing should be targeted at developers/implementers and for them the raw codepoint ranges with no additional context are IMO the most convenient.

Reply:
@hukkin, I'm ok either way. I've meanwhile updated the text with the comments from other reviewers. If there's consensus I can change it if need be.

Reply:
This has now been reviewed again by @ChristianSi (who made a few suggestions w.r.t. formatting, which has been committed), so I'm closing this discussion.